Mathlib/Topology/FiberBundle/Trivialization.lean

```diff
@@ -265,6 +265,87 @@ theorem apply_mk_symm (e : Pretrivialization F (π F E)) {b : B} (hb : b ∈ e.b
 
 end Zero
 
+open scoped Classical in
+/-- The restriction of a pretrivialization to a subset of the base. The argument `c` ensures that
+`proj ⁻¹' s` is nonempty when `s` and `F` are both nonempty, and supplies junk values. -/
+noncomputable def restrictPreimage' (e : Pretrivialization F proj) (s : Set B)
+    (c : s × F → proj ⁻¹' s) : Pretrivialization F (s.restrictPreimage proj) where
```

Comment:
Do we really want to have the power to control the junk values? If not, how about just:
```suggestion
    (ne : (proj ⁻¹' s).Nonempty) : Pretrivialization F (s.restrictPreimage proj) where
```
and then `else ⟨ne.some, Nonempty.some_mem ne⟩` instead of `else c x` below?